### PR TITLE
Add proxy for apiserver

### PR DESCRIPTION
**Pull request by steveodonovan**
# Changes

https://github.com/tektoncd/dashboard/issues/267

Adds a proxy for the api server

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by steveodonovan** [comment]
/assign @a-roberts @jessm12 

Could I get a review please ? Verified on minishift. I also updated the e2e tests. Looks like there were a few issues there, it was pointing to a [private repo](https://github.com/ncskier/go-hello-world/) and applying a [deployment and service](https://github.com/ncskier/go-hello-world/tree/master/config).

File: README.md [hunk 1]
```diff
@@ -39,11 +39,11 @@ The `install-dev.sh` script will build and push an image of the Tekton Dashboard
 4. Install build config `$oc process -f config/templates/build.yaml | oc apply -f-`
 
 ## Accessing the Dashboard
-The Dashboard has can be accessed through its ClusterIP Service by running `kubectl proxy`. Assuming default is the install namespace for the dashboard, you can access the web UI at `localhost:8001/api/v1/namespaces/default/services/tekton-dashboard:http/proxy/`
+The Dashboard has can be accessed through its ClusterIP Service by running `kubectl proxy`. Assuming default is the install namespace for the dashboard, you can access the web UI at `localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/`
```
Comment (a-roberts) [inline README.md]: `has can`, lingering from before

Assuming `default` is the install namespace should now be `assuming tekton-pipelines`

File: config/dev-tekton-dashboard.yaml [hunk 2]
```diff
@@ -1,8 +1,53 @@
+# ------------------- Dashboard Service Account ------------------- #
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    k8s-app: tekton-dashboard
```
Comment (a-roberts) [inline config/dev-tekton-dashboard.yaml]: Why `k8s-app` and not `app` like we do for the dashboard? FWIW either is fine I'd just rather us be consistent

File: test/e2e-tests.sh [hunk 3]
```diff
@@ -31,39 +31,10 @@ install_dashboard_backend
...
+kubectl port-forward $(kubectl get pod --namespace tekton-pipelines -l app=tekton-dashboard -o name)  --namespace tekton-pipelines 9097:9097 &
 dashboardForwardPID=$!
 
-#Apply permissions to be able to curl endpoints 
-kubectl apply -f $tekton_repo_dir/test/test-rbac.yaml
-kubectl apply -f $tekton_repo_dir/test/kaniko-build-task.yaml
-kubectl apply -f $tekton_repo_dir/test/deploy-task-insecure.yaml
-kubectl apply -f $tekton_repo_dir/test/Pipeline.yaml
-
-#API configuration
```
Comment (a-roberts) [inline test/e2e-tests.sh]: Hmm, why remove the manual pipelinerun e2e test? This is still a supported thing - yeah it's deploying an application and a service, and a ksvc, but surely then we should just have another simple role set up for the tests only.

**Comment by steveodonovan** [comment]
@a-roberts Added the test back in with the roles. Nothing else should be needed given we have namespace selection now. Tested extensions and they work but have to be in the installed namespace (which is now tekton-pipelines)

File: config/release/gcr-tekton-dashboard.yaml [hunk 4]
```diff
@@ -1,3 +1,54 @@
...
+---
+# ------------------- Dashboard Role & Role Binding ------------------- #
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: tekton-dashboard-minimal
+rules:
+  - apiGroups: [""]
+    resources: ["serviceaccounts"]
+    verbs: ["get", "list", "update"]
```
Comment (a-roberts) [inline config/release/gcr-tekton-dashboard.yaml]: @carlos-logro FYI, when you add your secret patching you'll need to add the `patch` verb here - take note of this role Stephen is adding or you'll run into RBAC issues, I'm cool for the patch verb to be added as part of that functionality you're adding

**Comment by a-roberts** [comment]
/lgtm

We should next update the webhook-extension docs to include tekton-pipelines as the namespace in our API requests, that can be done independently of this. I'll create an issue for that

**Pull request merged**